DSharpPlus.Core/Rest/RestClient.cs

```diff
@@ -0,0 +1,79 @@
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+using DSharpPlus.Caching.Abstractions;
+using DSharpPlus.Core.Exceptions;
+
+using Microsoft.Extensions.Options;
+
+using Polly;
+using Polly.Wrap;
+
+namespace DSharpPlus.Core.Rest
+{
+    /// <summary>
+    /// Represents a client for all rest requests to Discord.
+    /// </summary>
+    public class RestClient
+    {
+        private readonly ICacheService __cache;
+        private readonly HttpClient __http_client;
+        private readonly AsyncPolicyWrap<HttpResponseMessage>? __wrapped_policy;
+
+        private readonly string __token;
```

Comment:
Tokens shouldn't live in the rest client.

Reply:
where should they live?

Reply:
:v I didn't say it would be used for other things. I said it made the client *rigid*. 

Look at Remora's method of creating rest requests; it's pretty nice to use.

Reply:
but not as nice to implement, if i recall correctly :v

yeah, we could put the token somewhere else, but as it stands i don't *really* see the practical use for that